ir/visitor.h

```diff
@@ -388,14 +389,12 @@ class Modifier : public virtual Visitor {
 #undef DECLARE_VISIT_FUNCTIONS
     void revisit_visited();
     bool visit_in_progress(const IR::Node *) const;
+    void visitOnce() const override;
+    void visitAgain() const override;
 };
 
 class Inspector : public virtual Visitor {
-    struct info_t {
-        bool done, visitOnce;
-    };
-    typedef std::unordered_map<const IR::Node *, info_t> visited_t;
-    std::shared_ptr<visited_t> visited;
+    std::shared_ptr<Tracker> visited;
```

Comment:
The point of having this be an info_t instead of a trscker is tha an Inspector can't change nodes, so only needs to store 2 bits per node visited and not an additional pointer.  If you change it to an hvec_map instead of an unordered_map, it would might be possible to make it even more compact.

Reply:
See next PRs for further improvements here :)

Reply:
And for now I cannot use it due to https://github.com/p4lang/p4c/issues/4389